src/sentry/reprocessing2.py

```diff
@@ -264,7 +264,68 @@ def reprocess_event(project_id, event_id, start_time):
     )
 
 
-def delete_old_primary_hash(event):
+def buffered_delete_old_primary_hash(event, force_flush_batch: bool = False):
```

Comment:
There's an invocation of `buffered_handle_remaining_events` with `force_flush_batch=True` that we'd somehow have to replicate. I don't have a concrete idea how to though, as at the point where reprocessing is finished, there's no cheap way to retrieve all the old_primary_hashes that you need to force_flush.

Why is this needed? The way the code works is:

1. pile up more and more items in redis
2. if the items are too many, "send a batch" (i.e. call a method on eventstream)

but what if there are never too many items? then we never send out a batch. so at the end of reprocessing we do another call with `force-flush-batch` that takes all items from redis and puts them into a batch, regardless of how many there are

Reply:
@relaxolotl re your comment about mocking an event:

your crash comes from the fact that `event` is assumed to be a `sentry.eventstore.models.Event` class in this function, and you're passing a dict.

but I would change this signature to take `primary_hash, group_id, project_id` explicitly, then you don't have to do any of that

Reply:
done.

the signature is actually a unfortunately little more complex than that, as a getter for the primary hash (`event.get_primary_key`) is invoked on the event itself. i was reluctant to use this approach because of it, but it looks like there's no real way around it.